tmt/utils.py

```diff
@@ -85,6 +85,9 @@
 # A "environment" type, representing name/value environment variables.
 EnvironmentType = Dict[str, str]
 
+# Workdir type, can be True, a string, a path or None
+WorkdirType = Union[Literal[True], str, None]
```

Comment:
It's the `workdir` parameter that's allowed to be `True` (and `str` and `None`), not the workdir "value" itself. Therefore calling it a "workdir type" seems misleading to me, because there is no such thing as "workdir is `True`". `workdir` as a parameter of `_workdir_load()` method, yes, but not the workdir as such.

I do agree it makes sense to have an alias, it seems to be used in several places, I'd just propose a different name. `WorkdirParameterType` or `WorkdirArgumentType` or something along these lines. `WorkdirType` is what `Common._workdir` is, and that's pretty much `Optional[str]` - no `True` is going to appear there because any `True` that enters workdir-related methods is turned into some kind of string sooner or later.

Reply:
:+1: Maybe introduce 2 aliases? One `WorkdirType = Optional[str]` and then `WorkdirArgumentType = Union[WorkdirType, Literal[True]]` and use the appropriate one based on the semantics?

Reply:
Since `WorkdirType` aka `Optional[str]` is probably going to be used just once, i.e. when annotating `Common._workdir`, I for one don't see much advantage in adding an alias for it. On the other hand, if `WorkdirType` would become more complicated than just a mere `Optional[str]`, e.g. during a - hypothetical, for the sake of argument - switch to `pathlib.Path` (`Optional[str, pathlib.Path]`), then I'd definitely see the advantage.

Reply:
thanks, makes sense, I can add both, it will be more readable